### PR TITLE
Improvements in fitting classes [Part 1]

### PR DESCRIPTION
**Pull request by xvallspl**
Part of the reverted fitting improvements PR that should not affect anyone

### CODE REVIEW

**Comment by xvallspl** [comment]
@phsft-bot build!

**Comment by xvallspl** [comment]
Just to be on the safe side I will not merge this PR until I have time to check with the people who saw the building errors before, probably next week (if not possible, after CSC)

File: math/mathcore/inc/Fit/EvaluateChi2.hxx [hunk 1]
```diff
@@ -398,6 +382,23 @@ namespace FitUtil {
          Error("FitUtil::EvaluateChi2<T>::EvalResidual", "The vectorized evaluation of the Chi2 with the ith residual is still not supported");
          return -1.;
       }
+
+       // Compute a mask to filter out infinite numbers and NaN values.
+      // The argument rval is updated so infinite numbers and NaN values are replaced by
+      // maximum finite values (preserving the original sign).
+      static vecCore::Mask<T> CheckInfNaNValues(T &rval)
+      {
+         auto mask = rval > -vecCore::NumericLimits<T>::Max() && rval < vecCore::NumericLimits<T>::Max();
```
Comment (amadio) [inline math/mathcore/inc/Fit/EvaluateChi2.hxx]: Please use `vecCore::NumericLimits<T>::Lowest()`.

**Comment by xvallspl** [comment]
I still intend to try and merge this back in when I have a moment to retake it!

**Comment by lmoneta** [comment]
@xvallspl  Thank you avid if you can do it. It would be great to close this one

File: hist/hist/inc/TF1.h [hunk 2]
```diff
@@ -789,7 +789,8 @@ inline T TF1::EvalParTempl(const T *data, const Double_t *params)
 }
 
 #ifdef R__HAS_VECCORE
-// Internal to TF1. Evaluates Vectorized TF1 on data of type Double_v
+
+// Internal to TF1. Evaluates Vectorized TF1 on data of type Double_t
 inline double TF1::EvalParVec(const Double_t *data, const Double_t *params)
```
[the remark this answers is not in the document]
Comment (xvallspl) [reply hist/hist/inc/TF1.h]: Checked that it doesn't break compilation :ok_hand:

**Comment by xvallspl** [comment]
While fixing and old bug I used TError for the early termination of the function, and clang-tidy is not picking that up... Should I unnecesarily return some double as an error code or should I just ignore clang tidy? @lmoneta @Axel-Naumann ?

**Comment by Axel-Naumann** [comment]
To save time - could you point me to the file / line where you use this construct?

**Comment by xvallspl** [comment]
Sure! https://github.com/xvallspl/root/blob/f48ef95a6158dcf75b852a856697d4f6921340e6/math/mathcore/inc/Fit/FitUtil.h#L144

**Comment by Axel-Naumann** [comment]
`Error` doesn't abort, it just prints an error message. Please communicate the error through a return value, e.g. by returning `nan`.

**Comment by xvallspl** [comment]
Hah, sorry for that. I was convinced it terminated as well.

**Comment by pcanal** [comment]
> Hah, sorry for that. I was convinced it terminated as well.

The looging routine that terminates is 'Fatal' rather than 'Error'

**Comment by lmoneta** [comment]
Hi Xavi, 
 Can you remind me please what this PR brings in addition ? 
I see a re-arrange of the functions implementing the fit methods, but  in addition what else was changed ? The commits message are short

**Comment by xvallspl** [comment]
It was the grand refactoring. I can just close it and come back to it maybe later if so.

**Comment by lmoneta** [comment]
Don;t understand why adding a comment trigger a re-build even if the PR has a conflict. It should not do that. 

A part from the re-factoring, which I agree is a nice thing to do but not urgent, do you remember if there are other fixes worth merging ? 

If not maybe then it is probably better to close it.  

Lorenzo

**Pull request closed without merging**